### PR TITLE
Nfdiv 1195 confirm service bailiff refactor

### PR DESCRIPTION
**Pull request by mark-butler-solirius**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1195

### Change description ###

Functional Tests for NFDIV-894 and refactor of getOrderSummary to parameterised getOrderSummaryByServiceEvent

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseWorkerAlternativeServicePaymentFT.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Map<String, Object> caseData = caseData(REQUEST);
+
+        caseData.put("alternativeServiceType", "deemed");
+
+        Response response = triggerCallback(caseData, CASEWORKER_SERVICE_PAYMENT,
+            "/callbacks/about-to-start?page=alternativeServicePayment");
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        System.out.println(response.asString());
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseWorkerAlternativeServicePaymentFT.java:36]: sysout can be removed

File: src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseWorkerAlternativeServicePaymentFT.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        Map<String, Object> caseData = caseData(REQUEST);
+
+        caseData.put("alternativeServiceType", "bailiff");
+
+        Response response = triggerCallback(caseData, CASEWORKER_SERVICE_PAYMENT,
+            "/callbacks/about-to-start?page=alternativeServicePayment");
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
+
+        System.out.println(response.asString());
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseWorkerAlternativeServicePaymentFT.java:58]: same here

File: src/functionalTest/resources/responses/response-caseworker-confirm-payment-bailiff.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "applicant1FirstName":"John",
+    "applicant1LastName":"Smith",
+    "applicant1Email":"simulate-delivered@notifications.service.gov.uk",
+    "applicant2Gender":"female",
+    "solServiceTruthStatement":"I believe that the facts stated in the application are true.",
+    "solPaymentHowToPay":"feePayByAccount",
+    "alternativeServiceType":"bailiff",
+    "servicePaymentFeeOrderSummary": {
+      "PaymentReference":null,
+      "PaymentTotal":"27500",
```
Comment (nitinprabhu) [inline src/functionalTest/resources/responses/response-caseworker-confirm-payment-bailiff.json:13]: I think we need to set this in pounds. @linusnorton I remember you had to change the amount to pounds for PBA do we need to do same here

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAlternativeServicePayment.java [hunk 4]
```diff
@@ -95,4 +95,22 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
...
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("Submit application about to submit callback invoked");
+
+        CaseData data = details.getData();
+        State state = details.getState();
+
+        log.info("beforeDetails state {}", beforeDetails.getState());
+
+        log.info("afterDetails state {}", details.getState());
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAlternativeServicePayment.java:109]: do we need this ?

File: src/functionalTest/resources/responses/response-caseworker-confirm-payment-bailiff.json (hunk 3 above)
Comment (mark-butler-solirius) [reply src/functionalTest/resources/responses/response-caseworker-confirm-payment-bailiff.json:13]: The CreditAccountPaymentRequest.setamount method is still dividing the ordersummary payment total by 100

**Pull request merged**